core/src/main/scala/org/apache/spark/deploy/security/ServiceCredentialProvider.scala

```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.spark.deploy.yarn.security
+package org.apache.spark.deploy.security
 
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.security.{Credentials, UserGroupInformation}
```

Comment:
@vanzin Would be good to keep backward compatibility since there are users of the interface.
I did not realize yarn was excluded from MIMA - that is unfortunate.

Cursory look at ServiceLoader seemed to indicate it might be very involved to provide backward compatibility (`yarn` interface extends from `core`, load for both interfaces; and use both as `core` impl will do ? or did I miss something here ?)